### PR TITLE
Add article for RFC22

### PR DESCRIPTION
**Pull request by Leont**
Sorry I'm late. If you have a little time I can still process feedback. I'm not entirely satisfies with it yet, but EOUTOFTIME

### CODE REVIEW

File: 20th/articles/rfc22.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# RFC22: Control flow: Builtin switch statement, by Damian Conway
+
```
Comment (JJ) [inline 20th/articles/rfc22.md:2]: Please remember to link the RFC somewhere in the body

File: 20th/articles/rfc22.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+## The problem
+
+C has switch/case, and many other languages either copied it, or created a similar construct. Perl in 2000 didn't have any such thing, and this was seen as a lack.
+
+## A Tale of Two Languages
+
+This RFC not only became two (related) features, it did so in both Perl and Raku with dramatically different results: in Perl it's widely considered the biggest design failure of the past two decades, whereas in Raku it's an entirely non-controversial. The switch that Perl ended up with is working very similar to the original proposal. This is actually helpful in analysing what changes were necessary to make it a successful feature. It looks something like this (in both languages):
+
```
Comment (JJ) [inline 20th/articles/rfc22.md:10]: Try and link the documentation somewhere around here if possible.

File: 20th/articles/rfc22.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		say "One!";
+	}
+	when $_ > 42 {
+		say "It's more than life, the universe and everything";
+	}
+}
+
+~~~
+
+The switch is actually two features (for the price of only one RFC): smartmatch and given/when on top of it. Smartmatch is an operator `~~` that checks if the left hand side fits the constraints of the right hand side. Given/when is a construct that smartmatches the given argument to a series of when arguments (e.g. `$given ~~ $when`) until one succeeds.
```
Comment (JJ) [inline 20th/articles/rfc22.md:29]: Maybe link here both features.

File: 20th/articles/rfc22.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+However, one of the distinguishing features of Perl is that it doesn't generally overload operators, instead it has different operators for different purposes (e.g. `==` for comparing numbers and `eq` for comparing strings). Smartmatch however is inherently all about overloading. This mismatch is essentially the source of all the trouble of smartmatch in Perl. Raku on the other hand has an extensive type-system, and is not so dependent on type specific operators (though it still has some for convenience), and hence is much more predictable.
+
+Most obviously, the type system of Raku means that it doesn't use a table of possibilities, but instead `$left ~~ $right` just translates to `$right.ACCEPTS($left)`. The right-first semantics makes it a lot easier to reason about (e.g. matching against a number will always do numeric equality).
+
+It means it can easily distinguish between a string and an integer, unlike Perl which has to guess what you meant: `$foo ~~ 1` always means `$foo == 1`, and `$foo ~~ "foo"` always means `$foo eq "foo". In Perl, `$foo ~~ "1"` would do a numeric match.
+
```
Comment (JJ) [inline 20th/articles/rfc22.md:36]: close code enclosure here

**Review by JJ — APPROVED** [review]
Very nice article, please consider suggestions and check errors. Also tell me if you want me to upload it, in which case I'll need to give you permissions in WP.

File: 20th/articles/rfc22.md (hunk 1 above)
Comment (Leont) [reply 20th/articles/rfc22.md:2]: D'oh!

File: 20th/articles/rfc22.md (hunk 3 above)
Comment (Leont) [reply 20th/articles/rfc22.md:29]: Yeah, that's probably a good idea

**Comment by Leont** [comment]
>  @JJ merged commit 7172ad1 into Raku:master 17 minutes ago 

Was that on purpose?

> Also tell me if you want me to upload it, in which case I'll need to give you permissions in WP.

I guess I should put the updated version there instead?

**Comment by JJ** [comment]
El mar., 11 ago. 2020 a las 9:10, Leon Timmermans (<notifications@github.com>)
escribió:

> @JJ <https://github.com/JJ> merged commit 7172ad1
> <https://github.com/Raku/advent/commit/7172ad14707e168dc3e63ecf3b24d0d987011c7f>
> into Raku:master 17 minutes ago
>
> Was that on purpose?
>

Yep; this is not a production repo, after all. It's a nice article :-)

Also tell me if you want me to upload it, in which case I'll need to give
> you permissions in WP.
>
> I guess I should put the updated version there instead?
>

Correct. You have privs?

**Pull request merged**